sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/ParseFunction.scala

```diff
@@ -17,37 +17,49 @@
 
 package edu.usc.irds.sparkler.pipeline
 
-import java.io.ByteArrayInputStream
+import java.io.{InputStream, ByteArrayInputStream}
 
 import edu.usc.irds.sparkler.base.Loggable
 import edu.usc.irds.sparkler.model.CrawlData
-import org.apache.tika.metadata
+import org.apache.commons.io.IOUtils
+import org.apache.tika.metadata.Metadata
 import org.apache.tika.parser.AutoDetectParser
-import org.apache.tika.sax.LinkContentHandler
+import org.apache.tika.sax.{WriteOutContentHandler, BodyContentHandler, LinkContentHandler}
 
 import scala.collection.JavaConverters._
 
 /**
   * Created by thammegr on 6/7/16.
   */
-object ParseFunction extends ((CrawlData) => Set[String]) with Serializable with Loggable {
+object ParseFunction extends ((CrawlData) => (String, Set[String], Metadata)) with Serializable with Loggable {
```

Comment:
@karanjeets 
Lets refine the return type of this function.
How about creating a data type(POJO class) that include all these three as attributes?

### Suggestion:
Define a datatype
```scala
class ParserData (title:String, links:Set[String], meta: Metadata) {}
```
Then, modify function signature to this:
```scala
ParseFunction extends ((CrawlData) => (ParseData)) with Serializable with Loggable {...}
```